### PR TITLE
Slack DM - Reply in thread

### PR DESCRIPTION
**Pull request by liorblob**
## Status
Ready

## Description
Allow replying to the thread of the DM with the Demisto app.

## Screenshots
gif

## Required version of Demisto
5.0

## Does it break backward compatibility?
Changes the way users reply to questions from the Demisto app.

## Must have
- [X] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

File: Integrations/Slack/Slack.py [hunk 1]
```diff
@@ -700,25 +702,59 @@ def check_for_mirrors():
...
-        if len(id_and_task) > 1:
-            task_id = id_and_task[1]
-        content = text.replace(entitlement, '', 1)
+        content, guid, incident_id, task_id = await extract_entitlement(entitlement_match.group(), text)
         demisto.handleEntitlementForUser(incident_id, guid, user.get('profile', {}).get('email'), content, task_id)
 
         return True
+    else:
+        integration_context = demisto.getIntegrationContext()
+        questions = integration_context.get('questions')
```
Comment (Itay4) [inline Integrations/Slack/Slack.py]:
```suggestion
        questions = integration_context.get('questions', [])
```

**Comment by liorblob** [comment]
@Itay4 Also pushed a fix for multiline JSON in DM. Release notes were not cleared from last release so I can't update the change log ATM.

**Comment by Itay4** [comment]
Ok let’s wait with the merge until after the clean anyway

File: Integrations/Slack/Slack.py [hunk 2]
```diff
@@ -945,22 +952,30 @@ def send_message(destinations: list, entry: str, ignore_add_url: bool, integrati
             bot_id = get_bot_id()
         for dest in destinations:
             invite_users_to_conversation(dest, [bot_id])
-        response = send_message_to_destinations(destinations, message, thread_id)
+        response = send_message_to_destinations(destinations, message, thread_id, blocks)
     return response
 
 
-def send_message_to_destinations(destinations: list, message: str, thread_id: str) -> dict:
+def send_message_to_destinations(destinations: list, message: str, thread_id: str, blocks: str) -> dict:
```
Comment (Itay4) [inline Integrations/Slack/Slack.py]: shouldnt we add default value to blocks here?

**Comment by liorblob** [comment]
@kirbles19 Please review changelog

**Comment by kirbles19** [comment]
@liorblob All done.

**Pull request merged**